aten/src/ATen/native/sparse/SparseBlas.cpp

```diff
@@ -172,6 +185,8 @@ void sparse_sampled_addmm_check_inputs(
       result.layout() == kSparseCsr,
       "sampled_addmm: Expected result to have sparse csr layout, but got ",
       result.layout());
+  TORCH_CHECK(result.dense_dim() == 0,
+      "sampled_addmm: Expected non-hybrid result tensor");
```

Comment:
I read "Expected non-hybrid result tensor" as "support for non-hybrid results not implemented yet", so it does not matter much if the exception is raised here or later after checking other inputs, it should fail anyway with hybrid `result`.

It could only matter if there exist tests that check the exception messages of invalid inputs but it is very likely that these tests would not use hybrid `result`.

That said, I would not mind moving the check further below where other checks on `result` are implemented.

Reply:
Just a bit confusing to the user to read about `result` which is not supplied. And we could save on memory allocations if the inputs are already incorrect.

Reply:
So, we need this check for both `self` and `result`.

Reply:
Oh thanks for the reminder! Updated!